### PR TITLE
Problem: No way to change default packet timeout for `icaauth` module

### LINKED ISSUE #731: Problem: Undecided default `TimeoutDuration` for interchain account packets
Currently, the default `TimeoutDuration` for interchain account packets (when sending a transaction to host chain on behalf of interchain account) is 5 minutes (which can be overridden by the caller). Since interchain accounts use ordered channels, the channel will close if any packet times out. 5 minutes is a safe duration for relaying packets.

There are four options:

1. Keep default timeout to 5 minutes
2. Change default timeout to some other value
3. Remove the default timeout and make it mandatory for the caller to provide it in request.
4. Add `DefaultTimeoutDuration` to `icaauth` params so that it can be configured and changed in future.

@tomtau @yihuang @calvinaco Please provide your inputs in comments.

### PR DESCRIPTION
**Pull request by devashishdxt**
Solution: Added `MinTimeoutDuration` to `icaauth` module parameters. Fixes #731.

- Also changed the protobuf generation code (it is now completely in-sync with cosmos-sdk)

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -755,6 +756,7 @@ func initParamsKeeper(appCodec codec.BinaryCodec, legacyAmino *codec.LegacyAmino
 	paramsKeeper.Subspace(ibchost.ModuleName)
 	paramsKeeper.Subspace(icacontrollertypes.SubModuleName)
 	paramsKeeper.Subspace(icahosttypes.SubModuleName)
+	paramsKeeper.Subspace(icaauthmoduletypes.ModuleName)
```
Comment (tomtau) [inline app/app.go:760]: I guess the upgrade handler should set this param to something?

File: app/docs/swagger-ui/swagger.yaml [hunk 2]
```diff
@@ -4620,6 +4620,469 @@ paths:
...
+
+                        {
+                          "@type": "type.googleapis.com/google.profile.Person",
+                          "firstName": <string>,
+                          "lastName": <string>
+                        }
+
+                    If the embedded message type is well-known and has a custom
+                    JSON
+
```
Comment (tomtau) [inline app/docs/swagger-ui/swagger.yaml:5012]: maybe swagger.yaml can be patched in a separate PR?

File: proto/icaauth/v1/query.proto [hunk 3]
```diff
@@ -4,21 +4,36 @@ package chainmain.icaauth.v1;
...
 import "cosmos/base/query/v1beta1/pagination.proto";
+import "icaauth/v1/params.proto";
 
 option go_package = "github.com/crypto-org-chain/chain-main/x/icaauth/types";
 
 // Query defines the gRPC querier service.
 service Query {
-    // InterchainAccountAddress queries the interchain account address for given `connectionId` and `owner`
-    rpc InterchainAccountAddress(QueryInterchainAccountAddressRequest) returns (QueryInterchainAccountAddressResponse) {
-        option (google.api.http).get = "/devashishdxt/mars/icaauth/interchain_account_address/{connectionId}/{owner}";
```
Comment (tomtau) [inline proto/icaauth/v1/query.proto:14]: oops, this one slipped the previous review

File: x/supply/types/query.pb.go [hunk 4]
```diff
@@ -6,6 +6,7 @@ package types
 import (
 	context "context"
 	fmt "fmt"
+	github_com_cosmos_cosmos_sdk_types "github.com/cosmos/cosmos-sdk/types"
 	types "github.com/cosmos/cosmos-sdk/types"
```
Comment (tomtau) [inline x/supply/types/query.pb.go:10]: this is a bit strange that the types are duplicated? and why is the supply query updated?

**Review by tomtau — COMMENTED** [review]
what is the "swagger.yaml.bak" file for?

File: app/docs/swagger-ui/swagger.yaml (hunk 2 above)
Comment (tomtau) [reply app/docs/swagger-ui/swagger.yaml:5012]: or maybe ok in this PR

File: app/app.go (hunk 1 above)
Comment (devashishdxt) [reply app/app.go:760]: There are two ways a doing this AFAIK.

- Disable `InitGenesis` of a module and set params manually.
- Let the default params set by `InitGenesis`.

For interchain accounts, we are manually setting params (and disabling `InitGenesis`) because we want to modify the default params. But for `icaauth` we don't want to modify the default value of genesis params.

Comment (devashishdxt) [reply app/app.go:760]: I've added a test for this in `test_upgrade.py` by querying params after the upgrade.

File: app/docs/swagger-ui/swagger.yaml (hunk 2 above)
Comment (devashishdxt) [reply app/docs/swagger-ui/swagger.yaml:5012]: If it is difficult for you to review this, I can split it in two PRs?

File: x/supply/types/query.pb.go (hunk 4 above)
Comment (devashishdxt) [reply x/supply/types/query.pb.go:10]: Because this PR changes the protobuf code generation logic (it is now exactly same as `cosmos-sdk`).

**Comment by devashishdxt** [comment]
> what is the "swagger.yaml.bak" file for?

Removed

File: app/docs/swagger-ui/swagger.yaml (hunk 2 above)
Comment (tomtau) [reply app/docs/swagger-ui/swagger.yaml:5012]: it should be ok to review in this PR, but can do it in more depth tomorrow

File: app/app.go (hunk 1 above)
Comment (tomtau) [reply app/app.go:760]: ok

**Review by tomtau — APPROVED** [review]
